md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDAO.java

```diff
@@ -702,5 +725,155 @@ else  if (childProcess.getProcessType().getProcessTypeId() == 36){
         return processList;
     }
 
+public String securityCheck(Integer processId,String username,String action){
+    Session session = sessionFactory.openSession();
+    session.beginTransaction();
+    Process process = (Process) session.get(Process.class, processId);
+    Criteria criteria = session.createCriteria(UserRoles.class).add(Restrictions.eq("users.username", username));
+    List<UserRoles> userRoles = criteria.list();
+    List<String> userRolesNameList=new ArrayList<>();
+    String processCreater=process.getUserRoles().getRole();
+    for(UserRoles userRoles1:userRoles)
+    {
+        userRolesNameList.add(userRoles1.getRole());
+    }
 
+    session.getTransaction().commit();
+    session.close();
+    List<Integer> readList=new ArrayList<>();
+    readList.add(4);
+    readList.add(5);
+    readList.add(6);
+    readList.add(7);
+    List<Integer> writeList=new ArrayList<>();
+    writeList.add(2);
+    writeList.add(3);
+    writeList.add(6);
+    writeList.add(7);
+    List<Integer> executeList=new ArrayList<>();
+    executeList.add(1);
+    executeList.add(3);
+    executeList.add(5);
+    executeList.add(7);
+    if (process.getUsers().getUsername().equals(username))
+    {
+        switch (action) {
+            case "write": if (writeList.contains(process.getPermissionTypeByUserAccessId().getPermissionTypeId())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1067)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 12 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1151)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "write" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)